### PR TITLE
[Mu4] fix #327681 illegal repeatsegment crash

### PR DESCRIPTION
**Pull request by jeetee**
illegal repeatsegment inserted due to a certain open volta, jump and end repeat combination

Resolves: https://musescore.org/en/node/327681

MU4 version of PR #10158
Fixes interpretation order so the end repeat is covered by the open volta if a jump also exists in the same measure.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [N/A - patch applied according to what builds in Mu3] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/engraving/libmscore/repeatlist.cpp [hunk 1]
```diff
@@ -848,7 +855,10 @@ void RepeatList::unwind()
                     && ((*repeatListElementIt)->getRepeatCount() < (*repeatListElementIt)->measure->repeatCount())
                     ) {
                     // Honor the repeat
-                    push_back(rs);
+                    Q_ASSERT(rs != nullptr);
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/repeatlist.cpp:858]: Not sure we're still (supposed to be) using `Q_ASSERT()` in master?

**Review by Jojo-Schmitz — APPROVED** [review]

**Comment by jeetee** [comment]
We might want to update Q_ASSERTS to IF_ASSERT_FAILED as per https://github.com/musescore/MuseScore/wiki/Error-handling; but I've not included it here yet, as it would touch more in this file than relevant to the issue.
Perhaps that replacement could be done in a code sweep commit?

**Comment by Jojo-Schmitz** [comment]
Rebase needed. Issue came up again in https://musescore.org/en/node/335346

**Review by Jojo-Schmitz — APPROVED** [review]
While I don't claim to fully understand it all, the proposed change does make sense to me ;-)

**Comment by jeetee** [comment]
@cbjeukendrup would be nice to have this merged in before beta; one less crash to worry about then ;-)

**Comment by cbjeukendrup** [comment]
I've asked @vpereverzev to review this, since we're already entering extra careful mode :)

When it comes to asserts, we have two flavours:
- simple (C++ built-in?) `assert(…)`
- the `IF_ASSERT_FAILED` macro:
    ```c++
    IF_ASSERT_FAILED(condition) {
        // code that will be run when condition is unexpectedly false
        // (if asserts are enabled (debug build), then the assert would trigger an abort and this point won't be reached)
    }
    ```

**Comment by jeetee** [comment]
The Q_ASSERT was there more as a glorified debug breakpoint; the further handling of the scenario is the same whether the assert would've been hit or not.

**Review by vpereverzev — APPROVED** [review]

**Comment by jeetee** [comment]
ty!

**Pull request merged**